tests/Behat/WebtoolsCookieConsentContext.php

```diff
@@ -110,21 +108,41 @@ public function assertNoOembedIframeWithCckUsage(): void {
   }
 
   /**
-   * Checks that the CCK is loaded on the <HEAD> section of the page.
+   * Checks that the CCK JSON is present on the <HEAD> section of the page.
    *
-   * @Then the CCK javascript is loaded on the head section of the page
+   * @Then the CCK JSON is present on the head section of the page
    */
-  public function assertCckJsLoaded(): void {
-    $this->assertSession()->elementExists('css', "head > script[src^='" . OE_WEBTOOLS_COOKIE_CONSENT_BANNER_COOKIE_URL . "']");
+  public function assertCckJsonExists(): void {
+    if (!$this->findCookieConsentJson()) {
+      throw new \Exception(sprintf('No cookie consent json found.'));
+    }
   }
 
   /**
-   * Checks that the CCK is not loaded on the <HEAD> section of the page.
+   * Checks that the CCK JSON is NOT present on the <HEAD> section of the page.
    *
-   * @Then the CCK javascript is not loaded on the head section of the page
+   * @Then the CCK JSON is not present on the head section of the page
    */
-  public function assertNoCckJsLoaded(): void {
-    $this->assertSession()->elementNotExists('css', "head > script[src^='" . OE_WEBTOOLS_COOKIE_CONSENT_BANNER_COOKIE_URL . "']");
+  public function assertNoCckJsonExists(): void {
+    if ($this->findCookieConsentJson()) {
+      throw new \Exception(sprintf('Cookie consent json found.'));
+    }
+  }
+
+  /**
+   * Find the cookie consent kit JSON code.
+   */
+  protected function findCookieConsentJson(): ?array {
+    $scripts = $this->getSession()->getPage()->findAll("css", "script[type=\"application/json\"]");
+    /** @var \Behat\Mink\Element\NodeElement $script */
+    foreach ($scripts as $script) {
+      $json_value = json_decode($script->getText(), TRUE);
```

Comment:
@brummbar suggested we simplify this test and simply do if ($script->getText() === ‘{utility;: ckk}’)

Reply:
Also we probably want to do something similar to what we do here https://github.com/openeuropa/oe_media/pull/130/files#diff-ec52143cc0b2c879b2eb5b5d09efa55ac775b35996117513ed04239227f3f51bR73 to avoid random errors like we had in the past.

Reply:
> Also we probably want to do something similar to what we do here https://github.com/openeuropa/oe_media/pull/130/files#diff-ec52143cc0b2c879b2eb5b5d09efa55ac775b35996117513ed04239227f3f51bR73 to avoid random errors like we had in the past.

In fact, this step used for now in scenarios with disabled javascript so we shouldn't have for now any random errors related to smartloader behavrior.

Reply:
I think there might be value in getting ready for the future and supporting javascript tests as well.

Reply:
Ok. I slightly refactored the method.